Sources/Linq2DynamoDb.DataContext/TableDefinitionWrapper.cs

```diff
@@ -360,45 +369,120 @@ protected override void InitReader(ISupervisableEnumerable reader)
         /// <summary>
         /// Fills in and executes an update batch
         /// </summary>
-        private Task ExecuteUpdateBatchAsync(IDictionary<EntityKey, Document> addedEntities, IDictionary<EntityKey, Document> modifiedEntities, ICollection<EntityKey> removedEntities)
+        private Task ExecuteUpdateBatchAsync(IDictionary<EntityKey, Document> addedEntities, IDictionary<EntityKey, Document> modifiedEntities, IDictionary<EntityKey, Document> removedEntities)
         {
-            var batch = this.TableDefinition.CreateBatchWrite();
+            var dynamoDbUpdateTasks = AddEntriesAsync(addedEntities)
```

Comment:
I must've misread. Batching would certainly be better in the case where we aren't using optimistic locking.